packages/superset-ui-core/src/chart/registries/ChartBuildQueryRegistrySingleton.ts

```diff
@@ -1,7 +1,17 @@
-import { Registry, makeSingleton, OverwritePolicy, QueryContext } from '../..';
+import { Registry, makeSingleton, OverwritePolicy, QueryContext, SetDataMaskHook } from '../..';
 
 // Ideally this would be <T extends QueryFormData>
-type BuildQuery = (formData: any) => QueryContext;
+export type BuildQuery<T = any> = (
+  formData: T,
+  options?: {
+    hooks?: {
+      [key: string]: any;
+      setDataMask: SetDataMaskHook;
+      cachedChanges?: any;
+      setCachedChanges: (newChanges: any) => void;
+    };
```

Comment:
I think we should make this either more generic (just a generic `[key: string]: HandlerFunction) or optionally move these new hooks to the `Hooks` type: https://github.com/apache-superset/superset-ui/blob/15a718a6090668122a1b384e7335d5b9d0d22779/packages/superset-ui-core/src/chart/models/ChartProps.ts#L16-L30